registry/client/repository.go

```diff
@@ -519,21 +519,23 @@ func (ms *manifests) Get(ctx context.Context, dgst digest.Digest, options ...dis
 	if resp.StatusCode == http.StatusNotModified {
 		return nil, distribution.ErrManifestNotModified
 	} else if SuccessStatus(resp.StatusCode) {
-		if contentDgst != nil {
-			dgst, err := digest.Parse(resp.Header.Get("Docker-Content-Digest"))
-			if err == nil {
-				*contentDgst = dgst
-			}
-		}
 		mt := resp.Header.Get("Content-Type")
 		body, err := io.ReadAll(resp.Body)
 		if err != nil {
 			return nil, err
 		}
-		m, _, err := distribution.UnmarshalManifest(mt, body)
+		m, desc, err := distribution.UnmarshalManifest(mt, body)
```

Comment:
I'd like to have some eyes on this, as the original implementation of this was to use the digest _as known by the registry_ https://github.com/distribution/distribution/pull/1775. Maybe that was a poor choice of words in that PR, but;

- Is the unmarshalfunction guaranteed to return the same  digest?
- Would this defeat the purpose of the `Docker-Content-Digest` header (IIRC, it's also used to verify if the downloaded digest matches the digest as specified in the header?)


(Maybe I'm wrong on either of them; just want to be sure this gets the right eyes) 😅

Reply:
I've tested a few different repositories other than AWS ECR, and they seem to be returning the exact same SHA in the HTTP header and in the Descriptor.Digest. 

As I wasn't sure that this does exist in the descriptor for other older manifest/registry versions, I thought to keep the old behaviour as default to ensure it would not impact anything or anyone. 

Indeed at some point maybe it makes sense to deprecate the old header based behaviour, especially since it's an explicit docker reference, and I assume that the lib is supposed to start becoming more agnostic of specific CREs, but I'll leave that for someone else to decide down the line.

More eyes welcome for sure :)